### ISSUE TITLE
Buttons may position incorrectly

### ISSUE THREAD

**Issue by PKGeorgiev** [node-red/node-red-dashboard#249]
Recently I upgraded from 2.1.0 to 2.4.0. On my dashboard I have many charts and one button. When I resize the browser the dashboard elements do re-arrange. But the button is position on a strange location:

![image](https://user-images.githubusercontent.com/9254467/27740118-be3a4ca8-5da8-11e7-8851-5842ee43ac66.png)

Here is button's config:

![image](https://user-images.githubusercontent.com/9254467/27740210-1a81acd6-5da9-11e7-834c-5416d3bc031b.png)

![image](https://user-images.githubusercontent.com/9254467/27740233-3380cc94-5da9-11e7-8f83-fc4ddba5c006.png)

Additional info:

Running on RPi3
Linux 4.9.24-v7+ (raspbian)
node -v: v6.10.3
NodeRed 0.16
Dashboard v2.4
Tested on: IE11, MS Edge 38.14393.1066.0 (EdgeHTML 14.14393), Chrome 58.0.3029.110

**Comment by dceejay** [comment]
Hi I don't suppose that you can share the actual flow code (or the dashboard elements) with me can you ?

**Comment by PKGeorgiev** [comment]
@dceejay I can share my flows. Can I send them on the mail specified in dashboard's package.json?
Can you advise me how to export all flows at once? Because I can only see an option to select nodes and export them flow by flow (I have many flows)?

**Comment by dceejay** [comment]
in your /home/pi/.node-red directory will be a flows_{your hostname}.json file...
that is the complete flow (not including credentials... don't send creds)
yes please to dceejay    AT    gmail

**Comment by Nodi-Rubrum** [comment]
Still need examples of overlap issues?  Is this issue bubbling up to be addressed any time soon?  Moreover, is there any know workaround for when this happens?

**Comment by dceejay** [comment]
you don't show how they are misaligned - looks okay to me (not knowing what it should look like)... layout looks same on Mac - FF 80.0.1 - Chrome 86.0.4240.22 

But no - nowhere near top of stack. It's such a fundamental task to strip out the layout engine and look to replace with something more modern, that we may as well consider it alongside dropping Angular 1 - etc - aka moving to a completely new dashboard.

**Comment by Nodi-Rubrum** [comment]
I will add screen print when I can, but it is very similar to the example above, if I maximize the browser window, it is misaligned every time.  If keep the window smaller than maximum, it sometimes is misaligned.  Button draw on top over or out of position in reference to the parent owning tab, just like the example at the top of this issue.

So maybe when NR 2.0 or such is release type of thing?  Then something might address this?  If I understand the qualification you provided.

**Comment by dceejay** [comment]
releases of Dashboard and Node-RED are not tied. Dashboard is a set of nodes that works with Node-RED and is a separate project (that also happens to be under the node-red org within OpenJS).

**Comment by Nodi-Rubrum** [comment]
Fine, even though I am sure many don't parse that fact, the question still is applicable in context of our discussion, are we talking about a long time before any such design change for dashboard?  Do you think we are talking months or even years?

**Comment by dceejay** [comment]
OK - to be 100% clear... I am not intending to look at this issue in the near to medium term - so probably years rather than months, maybe never. But as an open source project we are more than happy for others to help dig in and fix it with a pull request. The file that probably needs fixing is src/components/ui-card-panel/ui-masonry.js.

**Comment by knolleary** [comment]
@Nodi-Rubrum as your flow includes an HTTP Request node that doesn't resolve (is it intended to be run on your local network?), it is impossible to know what you expect the page to look like and what it does look like when it goes wrong. The symptom may be similar to the image at the top of the issue, but without a specific example of a flow that recreates the issue, along with images so we know what the expected result is and what the issue looks like when it goes wrong, it's going to be very hard for anyone to make progress on this. If you are able to replace the HTTP Request node with some hardcoded example data so it is entirely self-container, and also add the images showing the expected/wrong views, that would certainly help however picks this issue up.

**Comment by Nodi-Rubrum** [comment]
Yup, it is a local page only.  I will try to create an example that is transportable.  But really my example is just that another example of what exactly  this issue qualifies.  My example is a bit more extreme if you will, since I have 3 tabs that are overlapping and the buttons draw above the hidden overlaps.
